lab04/lab04.cpp

```diff
@@ -0,0 +1,136 @@
+﻿#include <iostream>
+#include <iomanip>
+
+
+using namespace std;
+
+
+void Exit(void(*f)())
+	{
+		while (true)
+		{
+			string exit;
+
+			f();
+			cout << "\n Do you want exit? Y/N \n";
+			cin >> exit;
+			if (exit == "N")
+			{
+				continue;
+			}
+			else if (exit == "Y")
+			{
+				break;
+			}
+			else
+			{
+				cout << "Error\n";
+				break;
+			}
+		}
+	}
+void SelectAndSizeOfType()
+	{
+		string input;
+
+		cout << endl;
+		cin >> input;
+
+		if (input == "INT")
+		{
+			cout << sizeof(int) << " байт";
+			cout << endl;
+		}
+		else
+			if (input == "FLOAT")
+			{
+				cout << sizeof(float) << " байт";
+				cout << endl;
+			}
+			else
+				if (input == "DOUBLE")
+				{
+					cout << sizeof(double) << " байт";
+					cout << endl;
+				}
+				else
+					if (input == "CHAR")
+					{
+						cout << sizeof(char) << " байт";
+						cout << endl;
+					}
+					else
+					{
+						cout << endl << "Ошибка : Введено некоректное название типа" << endl;
+					}
+	}
+void DrawTriangle()
+	{
+		char ch,
+			space   = ' '; 
+		int pattern = -1;
+		int empty;
+
+		cout << "Enter char\n";
+		cin >> ch;
+		cout << "Enter size\n";
+		cin >> empty;
+	
+		for (int s = empty - 1; s > 0; s--)
+		{
+			
+			pattern += 2;
```

Comment:
You could define `pattern` variable inside `for(...)` statement

Reply:
He is should be defined  in this statement , since the pattern should extend their variable into two in all iterations of the loop.

Reply:
you could define two variables in for() statement